Patches/Live_30/Raw_SQL/01_Spark_Teleport_Devices.sql

```diff
@@ -0,0 +1,89 @@
+	VALUES ( 'def_spark_teleport_device_hersh', 1, 2052, @categoryFlags, @options, '', 1, 2, 50000, 0, 100, 'def_spark_teleport_device_hersh_desc', 1, 0, 0);
+END
+
+SET @baseeid = (SELECT TOP 1 eid FROM entities WHERE ename = 'base_TM')
+SET @options = CONCAT('#baseId=n', @baseeid)
+
+IF NOT EXISTS (SELECT 1 FROM entitydefaults WHERE definitionname = 'def_spark_teleport_device_nv')
+BEGIN
+	INSERT INTO entitydefaults (definitionname, quantity, attributeflags, categoryflags, options, note, enabled, volume, mass, hidden, health, descriptiontoken, purchasable, tiertype, tierlevel) 
+	VALUES ( 'def_spark_teleport_device_nv', 1, 2052, @categoryFlags, @options, '', 1, 2, 50000, 0, 100, 'def_spark_teleport_device_nv_desc', 1, 0, 0);
+END
+
+GO
+
+---- Add Spark Teleport Devices into Syndicate shop ----
+
+DECLARE @definition INT
+DECLARE @itemshop_preset INT
+
+SET @definition = (SELECT TOP 1 definition FROM entitydefaults WHERE definitionname = 'def_spark_teleport_device_hersh')
+SET @itemshop_preset = (SELECT TOP 1 id FROM itemshoppresets WHERE name = 'tm_preset_pve')
+
+IF NOT EXISTS (SELECT 1 FROM itemshop WHERE targetdefinition = @definition AND presetid = @itemshop_preset)
+BEGIN
+	INSERT INTO itemshop (presetid, targetdefinition, targetamount, tmcoin, icscoin, asicoin, credit, unicoin, globallimit, purchasecount, standing) VALUES
+	(@itemshop_preset, @definition, 1, NULL, NULL, NULL, 5000000, 300, null, 0, null)
+END
+
+SET @definition = (SELECT TOP 1 definition FROM entitydefaults WHERE definitionname = 'def_spark_teleport_device_nv')
+
+IF NOT EXISTS (SELECT 1 FROM itemshop WHERE targetdefinition = @definition AND presetid = @itemshop_preset)
+BEGIN
+	INSERT INTO itemshop (presetid, targetdefinition, targetamount, tmcoin, icscoin, asicoin, credit, unicoin, globallimit, purchasecount, standing) VALUES
+	(@itemshop_preset, @definition, 1, NULL, NULL, NULL, 5000000, 300, null, 0, null)
+END
+
+GO
+
+---- Add Hershfield as spark teleport location ----
+
+DECLARE @basedefinition INT
+DECLARE @baseeid INT
+
+SET @basedefinition = (SELECT TOP 1 definition FROM entitydefaults WHERE definitionname = 'def_public_docking_base_pelistal')
+
+SET @baseeid = (SELECT TOP 1 eid FROM entities WHERE ename = 'base_tm_pve_exp1')
+
+IF NOT EXISTS (SELECT 1 FROM charactersparkteleports WHERE characterid = 0 AND baseeid = @baseeid AND basedefinition = @basedefinition)
+BEGIN
+	INSERT INTO [charactersparkteleports] (characterid, baseeid, basedefinition, zoneid, x, y) VALUES
+	(0, @baseeid, @basedefinition, 8, 945.5, 788.5)
+END
+
+---- Add New Virginia as spark teleport location ----
+
+SET @baseeid = (SELECT TOP 1 eid FROM entities WHERE ename = 'base_TM')
+
+IF NOT EXISTS (SELECT 1 FROM charactersparkteleports WHERE characterid = 0 AND baseeid = @baseeid AND basedefinition = @basedefinition)
+BEGIN
+	INSERT INTO [charactersparkteleports] (characterid, baseeid, basedefinition, zoneid, x, y) VALUES
+	(0, @baseeid, @basedefinition, 8, 1080.5, 850.5)
```

Comment:
New Virginia zone ID is 0 (not 8). The corresponding server source code does not use this value. But I think that it is necessary to correct so that in the future there are no problems.

Reply:
It doesn't really matter as those 3 values aren't used during spark teleporting, but i'll fix